gui-lib/mred/private/wxme/editor-canvas.rkt

```diff
@@ -466,49 +500,78 @@
 
   (define/public (popup-for-editor b m) #f)
 
+  ;; Reduces an ordered set of scroll events into the next scroll
+  ;; position and then scrolls to it.
+  (define (process-scroll-events events)
+    (define-values (ox oy)
+      (get-scroll/values))
+
+    (define-values (x y)
+      (for/fold ([x ox]
+                 [y oy])
+                ([event (in-list events)])
+        (define code (send event get-key-code))
+        (define-values (new-x new-y)
+          (case code
+            [(wheel-up wheel-down)
+             (cond
+               [(and allow-y-scroll? (not fake-y-scroll?))
+                (values
+                 x
+                 (max (+ y
+                         (inexact->exact
+                          (floor
+                           (* wheel-amt
+                              (if (eq? code 'wheel-up)
+                                  (- (send event get-wheel-steps))
+                                  (send event get-wheel-steps))))))
+                      0))]
+
+               [else
+                (values #f #f)])]
+
+            [(wheel-left wheel-right)
+             (cond
+               [(and allow-x-scroll? (not fake-x-scroll?))
+                (values
+                 (max (+ x
+                         (inexact->exact
+                          (floor
+                           (* wheel-amt
+                              (if (eq? code 'wheel-left)
+                                  (- (send event get-wheel-steps))
+                                  (send event get-wheel-steps))))))
+                      0)
+                 y)]
+
+               [else
+                (values #f #f)])]))
+
+        (values (or new-x x)
+                (or new-y y))))
+
+    (when (and x y)
+      (do-scroll x y #t ox oy))
+
+    (collect-garbage 'incremental))
+
+  ;; Called by the scroll-timer whenever scrolling events need to be
+  ;; applied.  Roughly once every 33ms if the canvas is focused.
+  (define/public (on-scroll-flush)
+    (unless (null? scroll-events)
+      (process-scroll-events (reverse scroll-events))
+      (set! scroll-events null)))
```

Comment:
Probably better to reverse those two lines, just in case `do-scroll` somehow leads to a `yield`.